### PR TITLE
[frontend] Info Settings Section

### LINKED ISSUE #13: Port Figma into frontend
just do the skeleton screen first.
implement the interface first then implement the functionality.

## Check how we're doing
We will pr when a section is ready.

✔️: done and merged
⌛: pr and waiting for review
❌: not finished

### (A) Home & Error ✔️
- [x] / (index)
- [x] (401)
- [x] (402)
- [x] (403)

### (B) Discover & News ❌
- [x] /news/newsID
- [ ] /discover?pageNumber={pageNumber}

### (C) Login & Signup ✔️
- [x] /login
- [x] /signup

### (D) Info Settings ✔️
- [x] /user/info
- [x] /seller/info

### (E) ViewShop ✔️
- [x] /sellerID/shop
- [x] /seller/manageProducts
- [x] /seller/manageProducts/new
- [x] /seller/manageProducts/productID

### (F) Order ✔️
- [x] /buyer/order
- [x] /buyer/order/orderID
- [x] /seller/order/orderID
- [x] /seller/order

### (G) Cart ⌛
include cart view and the detail part
- [x] /buyer/cart 

### (H) Product ❌
- [ ] /product

###  (I) Coupon ✔️
- [x] /admin/manageCoupon
- [x] /admin/manageCoupon/couponID
- [x] /admin/manageCoupon/new
- [x] /seller/manageCoupon
- [x] /seller/manageCoupon/new
- [x] /seller/manageCoupon/couponID
- [x] /sellerCoupon
- [x] /coupon

### (J) Admin ManageUser ✔️
- [x] /admin/manageUser

### (K) Security ✔️
- [x] /user/security
- [x] /user/security/password
- [x] /user/security/manageCreditCard
- [x] /user/security/manageCreditCard/new

### (L) Search ❌
include search not found
- [ ] /search
- [ ] /search?shop={keyword}
- [ ] /search?keyword={keyword}

### (M) Report ❌
- [ ] /seller/report
- [ ] /seller/report/new
- [ ] /seller/report/reportID
- [ ] (the missing admin report)

### PR DESCRIPTION
**Pull request by UGisBusy**
Reeeeesolves #13 partially

### (D) Info Settings
- [x] [/user/info](http://localhost:5173/user/info)
  - navigate steps: person icon (nav bar) > Personal Info
  - this is made by @KatoToyoki 
- [x] [/seller/info](http://localhost:5173/user/seller/info)
  -  navigate steps: Sell (nav bar) > Shop Info
 
### Testing Approach (if unconventional)

- start the frontend server
  ```sh
  cd frontend
  npm run dev
  ```
- go to the new pages via steps above
- check and click every pixel if u want

### Additional context

- things haven't been implemented 
  - read and display the initial values of each field
  - make the json format of the forms

### CODE REVIEW

**Review by KatoToyoki — COMMENTED** [review]
i don't know why but if i `npm run build` will have the following warning

```warnings when minifying css:
▲ [WARNING] A nested style rule cannot start with "Col" because it looks like the start of a declaration [css-syntax-error]

    <stdin>:532:2:
      532 │   Col {
          │   ~~~
          ╵   :is(Col)

  To start a nested style rule with an identifier, you need to wrap the identifier in ":is(...)" to prevent the rule from being parsed as a declaration.
```

**Comment by UGisBusy** [comment]
> i don't know why but if i `npm run build` will have the following warning
> 
> ```
> ▲ [WARNING] A nested style rule cannot start with "Col" because it looks like the start of a declaration [css-syntax-error]
> 
>     <stdin>:532:2:
>       532 │   Col {
>           │   ~~~
>           ╵   :is(Col)
> 
>   To start a nested style rule with an identifier, you need to wrap the identifier in ":is(...)" to prevent the rule from being parsed as a declaration.
> ```

i actually soloved that in another PR.
~~angry sop man noise~~

**Review by KatoToyoki — DISMISSED** [review]
ok then it lgtm

File: frontend/src/pages/user/seller/info/index.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import { useState } from 'react';
+
+import TButton from '@components/TButton';
+import InfoItem from '@components/InfoItem';
+import { Col, Form, Row } from 'react-bootstrap';
+
+const SellerInfo = () => {
+  //TODO: read the initial value
+  const [visibility, setVisibility] = useState<boolean>(true);
+  const [shopName, setshopName] = useState<string>('');
```
Comment (NOOBDY) [inline frontend/src/pages/user/seller/info/index.tsx]:
```suggestion
  const [shopName, setShopName] = useState<string>('');
```

File: frontend/src/pages/user/seller/info/index.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    setVisibility(e.target.checked);
+  };
+
+  return (
+    <div>
+      <div className='title'>Shop info</div>
+      <hr className='hr' />
+      <Row style={{ margin: '2% 0% 2% 0% ' }}>
+        <Col xs={12} md={4} className='center_vertical'>
+          Visiblity
```
Comment (NOOBDY) [inline frontend/src/pages/user/seller/info/index.tsx]:
```suggestion
          Visibility
```

**Review by NOOBDY — CHANGES_REQUESTED** [review]

**Comment by UGisBusy** [comment]
oh no (fixed

**Review by KatoToyoki — DISMISSED** [review]
lgtm

**Comment by UGisBusy** [comment]
Will update ``app/tsx`` after #77 merges.

File: frontend/src/pages/user/seller/info/index.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      <div className='title'>Shop info</div>
+      <hr className='hr' />
+      <Row style={{ margin: '2% 0% 2% 0% ' }}>
+        <Col xs={12} md={4} className='center_vertical'>
+          Visibility
+        </Col>
+        <Col className='left'>
+          <Form.Check
+            type='checkbox'
+            id='visiblity_checkbox'
```
Comment (NOOBDY) [inline frontend/src/pages/user/seller/info/index.tsx]:
```suggestion
            id='visibility_checkbox'
```

**Review by NOOBDY — CHANGES_REQUESTED** [review]

File: frontend/src/pages/user/seller/info/index.tsx [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+              <Col xs={8} xl={7}>
+                <div style={SellerItemStyle}>
+                  {/* upload img */}
+                  <div className='center'>
+                    <input
+                      type='file'
+                      onChange={iconOnChange}
+                      style={{ display: 'none' }}
+                      ref={hiddenFileInput}
+                    />
```
Comment (KatoToyoki) [inline frontend/src/pages/user/seller/info/index.tsx:84]: currently, i can't upload new image for shop in `/user/seller/info`, is it normal?

Comment (UGisBusy) [reply frontend/src/pages/user/seller/info/index.tsx:84]: do u click the shop icon?

Comment (KatoToyoki) [reply frontend/src/pages/user/seller/info/index.tsx:84]: sorry my bad, no problem now

**Review by KatoToyoki — DISMISSED** [review]
if we put safari aside, it's okay now

File: frontend/src/pages/user/buyer/info/index.tsx [hunk 5]
```diff
@@ -1,21 +1,135 @@
...
+  height: '100px',
+  cursor: 'pointer',
+  boxShadow: '2px 4px 10px 2px rgba(0, 0, 0, 0.25)',
+};
+
+const labelStyle = {
+  fontSize: '16px',
+  fontWeight: '500',
+  color: '#FFFFFFAA',
+};
```
Comment (KatoToyoki) [inline frontend/src/pages/user/buyer/info/index.tsx]: color codes typically only have 6 digits perhaps there may be some keyboard errors.

File: frontend/src/pages/user/seller/info/index.tsx [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+  height: '100px',
+  cursor: 'pointer',
+  boxShadow: '2px 4px 10px 2px rgba(0, 0, 0, 0.25)',
+};
+
+const labelStyle = {
+  fontSize: '16px',
+  fontWeight: '500',
+  color: '#FFFFFFAA',
+};
```
Comment (KatoToyoki) [inline frontend/src/pages/user/seller/info/index.tsx]: color codes typically only have 6 digits perhaps there may be some keyboard errors.

**Review by KatoToyoki — CHANGES_REQUESTED** [review]
we can put Safari aside, and i'm okay if the following require changes are done

File: frontend/src/pages/user/seller/info/index.tsx (hunk 6 above)
Comment (UGisBusy) [reply frontend/src/pages/user/seller/info/index.tsx]: that's opacity tho.

Comment (NOOBDY) [reply frontend/src/pages/user/seller/info/index.tsx]: isn't the last two for alpha values

Comment (NOOBDY) [reply frontend/src/pages/user/seller/info/index.tsx]: I have no idea if CSS supports them though

Comment (KatoToyoki) [reply frontend/src/pages/user/seller/info/index.tsx]: oh i don't know css supports this either
if you want to set opacity with color, i think `rgba(255, 0, 0, 0.2);` is better but it's personal preference

Comment (UGisBusy) [reply frontend/src/pages/user/seller/info/index.tsx]: it works, but i can change it to rgba(255, 255, 255, 0.67)
image

**Review by KatoToyoki — APPROVED** [review]

**Review by NOOBDY — APPROVED** [review]

**Pull request merged**